device/core/devdoc/device_client_requirements.md

```diff
@@ -91,19 +91,27 @@ client.sendEvent(new Message('hello world'), print);
 **SRS_NODE_DEVICE_CLIENT_16_058: [** The `close` method shall immediately call the `closeCallback` function if provided and the transport is already disconnected. **]**
 
 #### sendEvent(message, sendEventCallback)
+#### sendEvent(outputName, message, sendEventCallback)
 The `sendEvent` method sends an event message to the IoT Hub as the device indicated in the constructor argument.
+If an ouptut name is provided then the message is routed to the given output.
 
 **SRS_NODE_DEVICE_CLIENT_05_002: [** The `sendEvent` method shall send the event (indicated by the `message` argument) via the transport associated with the Client instance. **]**
 
+**SRS_NODE_DEVICE_CLIENT_18_010: [** if `outputName` is provided, `sendEvent` shall populate `message.outputName` with this name before possing it to the transport. **]**
```

Comment:
>outputName [](start = 102, length = 10)

never a big fan of modifying objects that come from the user - can't we pass the outputName to the send method on the transport and let the transport annotate its transport-specific message object accordingly (if necessary?)